### DISCUSSION TITLE
Cannot reconnect. Please reload the window.

### DISCUSSION THREAD

**Discussion by joeithub** [coder/code-server#1947] [Q&A] [answered]
I installed the code-server with package `code-server-3.4.1-amd64.rpm` on centos 7.When I visit on browser ,it always show up 
`Cannot reconnect. Please reload the window `  dialog a few seconds  .I don't known what to do . I need help .

**Comment by joeithub** [comment]
and also show this on terminal:
(node:2785) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 exit listeners added to [ChildProcess]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
(node:2785) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 error listeners added to [ChildProcess]. Use emitter.setMaxListeners() to increase limit

**Comment by code-asher** [comment]
Could you post the full output of code-server?

**Comment by joeithub** [comment]
![lALPDgfLOt__MGrNA07NBNI_1234_846](https://user-images.githubusercontent.com/44569366/89747771-cb8a4a00-daf2-11ea-9d5a-c45f336468d9.png)
![lALPDh0cMrytxHd_zQHY_472_127](https://user-images.githubusercontent.com/44569366/89747783-d8a73900-daf2-11ea-9d87-be654febdd56.png)

**Comment by joeithub** [comment]
Keep reminding me to reload

**Comment by joeithub** [comment]
When I reload a few times,this show up:
info  Using config file ~/.config/code-server/config.yaml
info  Using user-data-dir ~/.local/share/code-server
info  code-server 3.4.1 48f7c2724827e526eeaa6c2c151c520f48a61259
info  HTTP server listening on http://0.0.0.0:8090
info    - No authentication
info    - Not serving HTTPS
events.js:292
      throw er; // Unhandled 'error' event
      ^
Error: read ECONNRESET
    at TCP.onStreamRead (internal/stream_base_commons.js:205:27)
Emitted 'error' event on Socket instance at:
    at emitErrorNT (internal/streams/destroy.js:100:8)
    at emitErrorCloseNT (internal/streams/destroy.js:68:3)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  errno: -104,
  code: 'ECONNRESET',
  syscall: 'read'
}
error VS Code exited unexpectedly with code 1

**Comment by joeithub** [comment]
![lALPDgQ9vDtZyl3NAZrNBLk_1209_410](https://user-images.githubusercontent.com/44569366/89748128-443dd600-daf4-11ea-9f47-2845f478e056.png)

**Comment by code-asher** [comment]
Hmmmm, that error is definitely not good but I don't think it's enough to pinpoint the issue.

Could you run code-server with the `--log debug` flag to get further debugging information? Then post the code-server logs again after trying to access it. It would also help to check the VS Code logs. Those can be found at `~/.local/share/code-server/logs/[date]`.

**Comment by Bob2b** [comment]
Hi, I have the exact same error. 
However I am using Freenas (freebsd 11.4 R2). So I am not sure it is the same problem.
Please let me know if I should raise a new ticket.

During installation with the script, two things where noticable:

1. the detected Freebsd version is different from the actual one.
2. Error, (unmet dependancy).
![afbeelding](https://user-images.githubusercontent.com/69627091/90412360-4d5b1400-e0ad-11ea-97e7-cefb96db0496.png)

During running of the code-server, The following logging was vissible.
(Used screenshot, because I had some trouble creating log files. Sorry about that.)
![afbeelding](https://user-images.githubusercontent.com/69627091/90412644-a75bd980-e0ad-11ea-8837-cf2ccefdcb67.png)
![afbeelding](https://user-images.githubusercontent.com/69627091/90412683-b17dd800-e0ad-11ea-9bcd-8275a09f4271.png)
![afbeelding](https://user-images.githubusercontent.com/69627091/90412713-bc386d00-e0ad-11ea-9540-65e47111a8b8.png)
![afbeelding](https://user-images.githubusercontent.com/69627091/90412760-ca868900-e0ad-11ea-8087-41f78db047b8.png)

The error in the webbrowser was visible during the last couple of lines of the last screenshot. (about T14:55:22).
Nog log files are present in the log folder.

**Comment by code-asher** [comment]
@Bob2b hmm yeah this might be a different issue. I've seen this error before when the native Node modules don't seem to install quite properly. It might be caused by missing libraries although I'm not sure why the install would succeed if that was the case. But it might be worth checking the build requirements to make sure you have everything: https://github.com/Microsoft/vscode/wiki/How-to-Contribute#prerequisites

If you `cd` into the `code-server` directory (`cd $(yarn global dir)/code-server`) and rebuild the modules it might fix it (`npm rebuild`).

**Comment by Bob2b** [comment]
@code-asher Oke thank you for the suggestions. I will look in to that.

**Comment by Bob2b** [comment, marked as answer]
FYI:
Indeed. Some missing dependancies. This solved it for me.
[https://github.com/cdr/code-server/issues/1826#issuecomment-651210655](https://github.com/cdr/code-server/issues/1826#issuecomment-651210655)

**Comment by code-asher** [comment]
Awesome, glad you sorted it out. I wonder if it would make sense for us to check for the required dependencies in the installation script.

**Comment by Bob2b** [comment]
You have got my vote 👍

**Comment by nhooyr** [comment]
I reopened https://github.com/cdr/code-server/issues/1826#issuecomment-651210655

**Comment by TristanBrotherton** [comment]
Im getting this same error - but I know why, I just don’t know how to fix.
If I run code-server and access it on my laptop - and leave that tab open, and then try to open it on my iPad I receive the same error. If I then close the session on my laptop the ipad session will then work.

My setup is running on Ubuntu Focal Fossil and going through Nginx per your setup guide.

**Reply by code-asher** [reply]
We have some buggy logic around concurrent connections: https://github.com/cdr/code-server/issues/1950

If the iPad is failing on the first connection though it's probably not related. If you're able to post the iPad browser logs please do so (I believe you have to connect it to a Mac).
